spec/request/posts_controller_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :request do
+  describe 'GET index' do
+    let(:user) { User.create(name: 'John Doe') }
+    let!(:post1) { Post.create(title: 'Post 1', text: 'Lorem ipsum', author: user) }
+    let!(:post2) { Post.create(title: 'Post 2', text: 'Dolor sit amet', author: user) }
+
+    it 'returns a successful response' do
+      get user_posts_path(user)
+      expect(response).to be_successful
+    end
+

```

Comment:
- [ ] Let's make sure the `index` and `show` actions for the `User` and `Post` controllers are tested. Kindly the `describe 'GET /show' do` of `Post` controllers ✍🏻 

- [ ] Make sure the User spec file include tests that ensures  `:show` view template is rendered. 
. Example: `expect(response).to render_template(:show)`